websocketOutportDriver/sender/webSocketSender.go

```diff
@@ -214,61 +196,40 @@ func (w *webSocketSender) waitForAck(remoteAddr string, counter uint64) {
 			return
 		}
 
-		_, ok = acksForAddress[counter]
+		ok = acksForAddress.ProcessAcknowledged(counter)
 		if ok {
-			w.removeAcknowledge(remoteAddr, counter)
 			return
 		}
 
 		time.Sleep(time.Millisecond)
 	}
 }
 
-func (w *webSocketSender) getAcknowledges() map[string]map[uint64]struct{} {
+func (w *webSocketSender) getAcknowledges() map[string]*websocketClientAcknowledgesHolder {
```

Comment:
Again, this is not properly fixed: 
Reason: you protected only the reading of the pointer of the map. Not the eventual read operation that can happen during the write on L124. Let's not return map stored inside structs. This is prone to concurrent accesses.
Strongly suggest to encapsulate more (mutAcknowledges + acknowledges map should be inside one struct that does one thing: correct management of the acknowledged counters)

Reply:
done -> extracted another componet